pkgs/development/libraries/qrcodegen/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+,
+}:
+let
+  version = "1.6.0";
+in
+stdenv.mkDerivation {
+  pname = "qrcodegen";
+  inherit version;
+
+  nativeBuildInputs = [ ];
+
+  buildInputs = [ ];
+
+  preBuild = "cd c";
+  installPhase = ''
+    mkdir -p $out/lib $out/include/qrcodegen
+    cp libqrcodegen.a $out/lib
+    cp qrcodegen.h $out/include/qrcodegen/
+  '';
+
+  src = fetchFromGitHub {
+    owner = "nayuki";
+    repo = "QR-Code-generator";
+    rev = "v${version}";
+    sha256 = "0iq9sv9na0vg996aqrxrjn9rrbiyy7sc9vslw945p3ky22pw3lql";
+  };
+
+  meta = {
+    description = "qrcode generator library in multiple languages";
+
+    longDescription = ''
+      This project aims to be the best, clearest library for generating QR Codes. My primary goals are flexible options and absolute correctness. Secondary goals are compact implementation size and good documentation comments.
+    '';
+
+    homepage = "https://github.com/nayuki/QR-Code-generator";
+
+    license = lib.licenses.mit;
+    platforms = lib.platforms.all;
+    maintainers = with lib.maintainers; [ mcbeth ];
```

Comment:
```suggestion
    license = licenses.mit;
    platforms = platforms.all;
    maintainers = with maintainers; [ mcbeth ];
```

Reply:
Thank you, these were all quite illustrative.  Thank you.